### PR TITLE
Commiting the swagger docs for list/query

### PR DESCRIPTION
**Pull request by SaniyaLalani**
added swagger docs for list/query

### CODE REVIEW

File: app/server.py [hunk 1]
```diff
@@ -76,6 +76,41 @@ def index():
 
 @app.route('/inventory', methods=[GET])
 def query_prod_info():
+    """
+    Retrieve a list of inventory items
+    This endpoint will return all details of the all the products in the inventory unless a query parameter is specificed
+    ---
+    tags:
+      - inventory
```
Comment (MingshiPeng) [inline app/server.py]: Please unify the indentation style, refer to swagger doc starting from line 139.

File: app/server.py [hunk 2]
```diff
@@ -76,6 +76,41 @@ def index():
 
 @app.route('/inventory', methods=[GET])
 def query_prod_info():
+    """
+    Retrieve a list of inventory items
```
Comment (MingshiPeng) [inline app/server.py]: Seems a little uninformative. Maybe something like "Query a list of inventory items based on given keywords"?
Line 81 is a little bit of confusing as well, "return all details of the all the products" is not the major functionality of this endpoint.

**Review by MingshiPeng — COMMENTED** [review]
It would be nice if you could provide a screenshot for the corresponding swagger doc. It might be easier to spot if something is missing or going wrong.

File: app/server.py (hunk 1 above)
Comment (SaniyaLalani) [reply app/server.py]: sure

File: app/server.py (hunk 2 above)
Comment (SaniyaLalani) [reply app/server.py]: hmm, I referred to the professor's notes. But sure I don't mind making it more informative

**Comment by MingshiPeng** [comment]
Also maybe connect this PR to the issue you are working on for easier future references?
And this endpoint does only "query". There is another endpoint for "list" specifically, so naming this PR "list/query" might be a little confusing.

**Comment by SaniyaLalani** [comment]
Actually list all is handled in the same function as query. Take a look at the code you'll understand

**Comment by MingshiPeng** [comment]
Oops, just checked the code, you're right. Sorry for the misunderstanding

**Comment by SaniyaLalani** [comment]
So I worked on both the issues. Just connected one to it. I could create a new issue to combine the two issues.

File: app/server.py [hunk 3]
```diff
@@ -76,6 +76,41 @@ def index():
...
+          description: Bad Request (invalid posted data)
+      200:
+        description: An array of all the products
+        schema:
+          type: array
+          items:
+            schema:
+              $ref: '#/definitions/Product'
+    """
     """ Query specific entries in the Inventory system """
```
Comment (MingshiPeng) [inline app/server.py]: Remove this line if you are replacing it with a new docstring.

**Comment by MingshiPeng** [comment]
Well... if it's too troublesome maybe just leave it. Connecting to one is fine. Not worth the additional trouble though.

**Comment by SaniyaLalani** [comment]
screen shot 2018-04-29 at 8 43 25 pm

**Comment by majia67** [comment]
As suggested by @MingshiPeng, it is better to also include a screenshot of the rendered Swagger doc (i.e. the "website" in the browser).

**Comment by SaniyaLalani** [comment]
screen shot 2018-04-30 at 5 09 10 pm
screen shot 2018-04-30 at 5 09 23 pm
screen shot 2018-04-30 at 5 09 34 pm

File: app/server.py [hunk 4]
```diff
@@ -76,7 +76,42 @@ def index():
...
+            in: query
+            description: the name of the product you are looking for
+            required: false
+            type: string
+      -     name: quantity
+            in: query
+            description: if you want to check how many products have a specfic quantity
+            required: false
+            type: integer
+      -     name: condition
```
Comment (jockyxu1122) [inline app/server.py]: Can you add an 'enum' attribute to "condition"? It would be clearer what exact conditions users can give.

**Review by majia67 — COMMENTED** [review]
LGTM. Please address the issue mentioned by @jockyxu1122 and wait him to approve.

File: app/server.py (hunk 4 above)
Comment (SaniyaLalani) [reply app/server.py]: So you want me to change the type to enum?

Comment (jockyxu1122) [reply app/server.py]: It's still a string. Just add an "enum" attribute.
enum:
  - new
  - used
  - open_boxed

**Comment by SaniyaLalani** [comment]
screen shot 2018-04-30 at 8 41 11 pm

**Review by majia67 — APPROVED** [review]
Ship it!

**Comment by majia67** [comment]
IBM Cloud [toolchain](https://console.ng.bluemix.net/devops/toolchains/d4dd8efb-da17-4b0c-b4e6-5599ec73dd60?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18"): [Delivery Pipeline](https://console.ng.bluemix.net/devops/pipelines/c71be43a-8963-4052-b97c-5da9b99de3f9?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18") deployed [inventory](https://github.com/NYUDevOpsInventory/inventory/releases/tag/deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-dev-20180501-185525 "deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-dev-20180501-185525") to [dev](https://console.bluemix.net/apps/1db994d8-c4f1-4076-85c0-7ff2674244e5?env_id=ibm:yp:us-south "ibm:yp:us-south:NYU_Devops_Inventory_Org:dev"), including commit 604b380491a841ca0dae8bc0a0b865eb70053ef1

**Comment by majia67** [comment]
IBM Cloud [toolchain](https://console.ng.bluemix.net/devops/toolchains/d4dd8efb-da17-4b0c-b4e6-5599ec73dd60?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18"): [Delivery Pipeline](https://console.ng.bluemix.net/devops/pipelines/c71be43a-8963-4052-b97c-5da9b99de3f9?env_id=ibm%3Ayp%3Aus-south "nyu-inventory-service-s18") deployed [inventory](https://github.com/NYUDevOpsInventory/inventory/releases/tag/deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-prod-20180501-185835 "deploy-ibm-yp-us-south-NYU_Devops_Inventory_Org-prod-20180501-185835") to [prod](https://console.bluemix.net/apps/e25e653b-a50f-4831-b93e-d8f9f484a484?env_id=ibm:yp:us-south "ibm:yp:us-south:NYU_Devops_Inventory_Org:prod"), including commit 604b380491a841ca0dae8bc0a0b865eb70053ef1

**Pull request merged**